solr/test-framework/src/java/org/apache/solr/cloud/MiniSolrCloudCluster.java

```diff
@@ -156,11 +157,12 @@ public class MiniSolrCloudCluster {
   private final boolean externalZkServer;
   private final List<JettySolrRunner> jettys = new CopyOnWriteArrayList<>();
   private final Path baseDir;
-  private final CloudSolrClient solrClient;
+  private CloudSolrClient solrClient;
   private final JettyConfig jettyConfig;
   private final boolean trackJettyMetrics;
 
   private final AtomicInteger nodeIds = new AtomicInteger();
+  private Map<String, CloudSolrClient> solrClientForCollectionCache = new ConcurrentHashMap<>();
```

Comment:
I suggest "solrClientByCollection" -- shorter and the "by" part is more typical (I think?) for maps to say what the key is.

Reply:
And can be final

Reply:
Done!